src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -134,6 +134,35 @@ public int getTotalUserCount(){
    * @return a list of messages received by the recipient, or empty list if the recipient
    * 	hasn't received message. List is sorted by time descending. 
    */
+
+  public List<Message> getAllMessages(){
+      List<Message> messages = new ArrayList<>();
+
+      Query query = new Query("Message")
+              .addSort("timestamp", SortDirection.DESCENDING);
+      PreparedQuery results = datastore.prepare(query);
+
+      for (Entity entity : results.asIterable()) {
+          try {
+              String idString = entity.getKey().getName();
+              UUID id = UUID.fromString(idString);
+              String user = (String) entity.getProperty("user");
+              String text = (String) entity.getProperty("text");
+              long timestamp = (long) entity.getProperty("timestamp");
+              String recipient = (String) entity.getProperty("recipient");
+
+              Message message = new Message(id, user, text, timestamp, recipient);
+              messages.add(message);
+          } catch (Exception e) {
+              System.err.println("Error reading message.");
```

Comment:
the System.err.println is unnecessary here.

I kind of think you actually want to throw the exception back out, don't you?  Under what conditions would you want to return a blank array if something went wrong?